torch/distributed/_tensor/ops/tensor_ops.py

```diff
@@ -598,3 +598,71 @@ def _update_schema_suggestion_for_cat(
         )
     ]
     return output_sharding
+
+
+@register_prop_rule(aten.split.Tensor)
+def split_rule(op_schema: OpSchema) -> OutputSharding:
+    output_spec_list: List[DTensorSpec] = []
+    input_spec = cast(DTensorSpec, op_schema.args_schema[0])
+    ndim = input_spec.ndim
+    split_size_or_sections = op_schema.args_schema[1]
+    dim = 0
+    if len(op_schema.args_schema) > 2:
+        dim = cast(int, op_schema.args_schema[2])
+    dim = normalize_dim(dim, ndim)
+
+    # TODO: just like slice op, split replicates before splitting
+    # on a sharded dimension
+    # TODO: shall we consider partial???
+    # TODO: consider splitting an empty tensor???
+    need_reshard = False
+    if is_tensor_dim_sharded(input_spec, dim=dim):
+        need_reshard = True
+        input_spec = DTensorSpec(
+            mesh=input_spec.mesh,
+            placements=unshard_tensor_dim(input_spec.placements, dim=dim),
+            shape=input_spec.shape,
+            ndim=input_spec.ndim,
+        )
+
+    if need_reshard:
+        return OutputSharding(
+            None,
+            schema_suggestions=[
+                OpSchema(
+                    func_schema=op_schema.func_schema,
+                    args_schema=(input_spec,) + op_schema.args_schema[1:],
+                    kwargs_schema={},
+                ),
+            ]
+        )
+
+    def size_split(N, i):
+        # Last chunk will be smaller if the tensor size N
+        # along the given dimension dim is not divisible by i.
+        assert i > 0
+        return [i] * (N // i) + ([N % i] if N % i != 0 else [])
```

Comment:
shouldn't you return `[i]` in the else branch?

Reply:
let's see the case where N = 6 and i = 3. The expected result of size splitting should be `[3, 3]`. `N // i == 2` and `N % i == 0`. So the first part `[i] * (N // i)` is `[3, 3]` and the second part should be `[]`. What do you think?

Reply:
Oh I see what this mean, I misread the "+" there.